src/tmp.jl

```diff
@@ -231,6 +231,54 @@ function tmp_create_constraints_indices(connection)
         WHERE main.type in ('hub', 'consumer')
         ",
     )
+
+    # -- The previous attempt used
+    # The idea below is to find all unique time_block_start values because
+    # this is uses strategy 'highest'. By ordering them, and making
+    # time_block_end[i] = time_block_start[i+1] - 1, we have all ranges.
+    # We use the `lead` function from SQL to get `time_block_start[i+1]`
+    # and row.num_timesteps is the maximum value for when i+1 > length
+    #
+    # The query below is trying to replace the following constraints_partitions example:
+    #= (
+            name = :highest_out,
+            partitions = _outflows,
+            strategy = :highest,
+            asset_filter = (a, y) -> graph[a].type in ["producer", "storage", "conversion"],
+        ),
+    =#
+    # The **highest** strategy is obtained simply by computing the union of all
+    # time_block_starts, since it consists of "all breakpoints".
+    # The time_block_end is computed a posteriori using the next time_block_start.
+    # The query below will use the WINDOW FUNCTION `lead` to compute the time
+    # block end.
+    # This query uses the assets and outgoing flows to compute partitions
+    # This will be useful when we have other `partitions` instead of `_allflows`
+    # SELECT asset, year, rep_period, time_block_start
+    # FROM asset_time_resolution
+    # UNION
+    DuckDB.execute(
+        connection,
+        "CREATE OR REPLACE TABLE cons_indices_highest_out AS
+        SELECT
+            main.asset,
+            main.year,
+            main.rep_period,
+            sub.time_block_start,
```

Comment:
```suggestion
            COALESCE(sub.time_block_start, 1) AS time_block_start,
```
I think this is the same issue that I had in #919 but for the benchmark tests, and that now is fixed in #920. The problem is that there are no outgoing flows (in this case), so `sub.time.block_start` doesn't exist. By using `COALESCE`, we make sure that if it doesn't exist, we use the range `1:num_timesteps`